extensions/prototypes/labgraph_monitor/src/components/SettingPanel/EdgeSettings.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {
+    Box,
+    Paper,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TableRow,
+    Typography,
+} from '@mui/material';
+import React from 'react';
+import { RootState } from '../../redux/store';
+import { useSelector } from 'react-redux';
+import WS_STATE from '../../redux/reducers/ws/enums/WS_STATE';
+
+/**
+ * A component that manages the settings of an edge.
+ * All components related to edge settings should be children of this component.
+ *
+ * @returns {JSX.Element}
+ */
+const Edge: React.FC = (): JSX.Element => {
+    const { selectedEdge } = useSelector((state: RootState) => state.config);
+    const { connection, graph: realtimeGraph } = useSelector(
+        (state: RootState) => state.ws
+    );
+    const { mockGraph } = useSelector((state: RootState) => state.mock);
+
+    const graph = connection === WS_STATE.CONNECTED ? realtimeGraph : mockGraph;
+
+    const messages =
+        graph && selectedEdge.target
+            ? graph['nodes'][selectedEdge.target]['upstreams'][
+                  selectedEdge.source
+              ]
+            : [];
+    return (
+        <React.Fragment>
+            <Box data-testid="edge-settings">
+                {messages.map((message, index) => {
+                    return (
+                        <TableContainer key={index} component={Paper}>
```

Comment:
It's not best practice, and you should see a warning for using `key={index}`, is there any other unique identifier we can use for the messages?

Reply:
Currently there is no unique value, that might be done at the backend at the time of serialization